### ISSUE TITLE
Feature Request - Update Policy Assignment Code to use parameters from api version 2022-06-01

### ISSUE THREAD

**Issue by birdnathan** [Azure/terraform-azurerm-caf-enterprise-scale#660]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### Is your feature request related to a problem?

resourceSelectors and Overrides are currently overlooked in all Policy Assignments as these are not present in: https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/main/resources.policy_assignments.tf

#### Describe the solution you'd like

I would like the ability to declare these as extensions to the existing or custom Policy Assignment files. Being able to add these will give us a nicer way of including/excluding resourceTypes from the Diagnostics Policy Set assignment for example:

```
"policyDefinitionId": "${root_scope_resource_id}/providers/Microsoft.Authorization/policySetDefinitions/Deploy-Diagnostics-LogAnalytics",
    "scope": "${current_scope_resource_id}",
    "enforcementMode": null,
    "resourceSelectors": [
      {
        "name": "exclude-storageAccounts",
        "selectors": [
          {
            "kind": "resourceType",
            "notIn": [
              "Microsoft.Storage/storageAccounts",
              "Microsoft.Storage/storageAccounts/blobServices",
              "Microsoft.Storage/storageAccounts/fileServices",
              "Microsoft.Storage/storageAccounts/tableServices",
              "Microsoft.Storage/storageAccounts/queueServices"
            ]
          }
        ]
      }
```

#### Additional context
This capability was made available in arm api [2022-06-01](https://learn.microsoft.com/en-us/azure/templates/microsoft.authorization/2022-06-01/policyassignments?pivots=deployment-language-terraform) and was also made available in [3.49.0](https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.49.0) of the azurerm tf provider.

**Comment by matt-FFFFFF** [comment]
Thanks @birdnathan We will add to our backlog internally for a future release

**Comment by matt-FFFFFF** [comment]
Hi @birdnathan 

We have this now in the module with v4

https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/commit/2467832c3af667f4226e94b09b9f0e1063f51779

let me know if this fixes your issue?

**Comment by birdnathan** [comment]
Half fixes the issue by the looks @matt-FFFFFF . Still need a dynamic resourceSelectors block, these are different to overrides.

**Comment by matt-FFFFFF** [comment]
fixed by #725
